odoo/addons/base/models/ir_rule.py

```diff
@@ -243,14 +244,26 @@ def get_record_description(rec):
         rules_description = '\n'.join(f'- {rule.name}' for rule in rules)
         failing_rules = _("Blame the following rules:\n%s", rules_description)
 
+        context = None
         if company_related:
-            failing_rules += "\n\n" + _('Note: this might be a multi-company issue. Switching company may help - in Odoo, not in real life!')
-
+            suggested_companies = records.sudo()._get_redirect_suggested_company()
+            if suggested_companies and len(suggested_companies) > 1:
+                resolution_info += "\n\n" + _('Note: this might be a multi-company issue. Switching company may help - in Odoo, not in real life!')
+            else:
+                suggested_company = suggested_companies if suggested_companies and suggested_companies in self.env.user.company_ids else None
+                if suggested_company:
+                    context = {'suggested_company': {'id': suggested_company.id, 'display_name': suggested_company.display_name}}
+                    resolution_info += "\n\n" + _('This seems to be a multi-company issue, you might be able to access the record by switching to the company: %s.', suggested_company.display_name)
+                else:
+                    resolution_info += "\n\n" + _('This seems to be a multi-company issue, but you do not have access to the proper company to access the record anyhow.')
```

Comment:
IMO, you can simplify these if and handle more correctly the case where there is no company_id ('This seems to be a multi-company issue, but you do not have access to the proper company to access the record anyhow.' => 'Note: this might be a multi-company issue. Switching company may help - in Odoo, not in real life!')

```suggestion
            if len(suggested_companies) != 1:
                resolution_info += "\n\n" + _('Note: this might be a multi-company issue. Switching company may help - in Odoo, not in real life!')
            elif suggested_companies in self.env.user.company_ids:
                context = {'suggested_company': {'id': suggested_company.id, 'display_name': suggested_company.display_name}}
                resolution_info += "\n\n" + _('This seems to be a multi-company issue, you might be able to access the record by switching to the company: %s.', suggested_company.display_name)
            else:
                resolution_info += "\n\n" + _('This seems to be a multi-company issue, but you do not have access to the proper company to access the record anyhow.')
```